_posts/2022-07-25-changelog-175.markdown

```diff
@@ -0,0 +1,129 @@
+
+* [#8868] Highlight macro-machinery-specific syntax in `macro_rules` definitions.
+  New highlighting should help to distinguish a syntax specific to macros from
+  an expansion code (by [@daym])
+
+  <img src="/assets/posts/changelog-175/macro-highlighting.png" alt="Macro highlighting" width="700px"/>
+
+* [#8869] Enable by default some inspections for dependencies in `Cargo.toml`
+
+    * `Invalid crate version` warns you about a non-existing crate version
+
+      <img src="/assets/posts/changelog-175/invalid-crate-version.png" alt="Invalid crate version" width="700px"/>
+
+    * `New crate version available` notifies you about a crate that could be updated.
+
+      {% include gif-img.html path="/assets/posts/changelog-175/new-crate-version-available" w="700px" %}
+
+      We tried to make the highlighting not too intrusive, because using an older version of a crate
+      may be quite normal. If you still don't like the inspection, you can simply [disable it][disable-inspection].
+
+* Improve detection of some errors:
+
+    * [#9026] Improve detection of duplicate definitions.
+      Add support for [`E0254`](https://doc.rust-lang.org/error-index.html#E0254),
+      [`E0259`](https://doc.rust-lang.org/error-index.html#E0259) and
+      [`E0260`](https://doc.rust-lang.org/error-index.html#E0260) errors
+
+    * [#9020] Detect [E0430](https://doc.rust-lang.org/error-index.html#E0430)
+      `The self import appears more than once in the list` compiler error
+
+    * [#9000] Detect [E0742](https://doc.rust-lang.org/error-index.html#E0742)
+      `Visibility is restricted to a module which isn’t an ancestor of the current item` compiler error
+
+* [#9024] Support `View | Type Info` action for field shorthand syntax.
+  You can invoke `Type Info` action using <kbd>Ctrl</kbd> + <kbd>shift</kbd> + <kbd>P</kbd> shortcut
+
+* [#7299] Fix debug pretty-printers to render `HashMap`/`HashSet` with MSVC LLDB on Windows
+
+## Performance Improvements
+
+* [#9021], [#9016], [#9015], [#8968] Slightly speed up type inference
+
+* [#8986] Speed up completion in procedural macros.
+  Please, note that procedural macro expansion is experimental. If you want to try it,
+  enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` experimental features
+
+## Fixes
+
+* [#8986] Try to fix up rust syntax before calling an attribute procedural macro placed on a function.
+  This fixes highlighting blinking during typing in a body of a function with a proc attribute macro on it
+
+* [#9090] Limit the maximal `#![recursion_limit]` attribute value to 1000 when expanding macros.
+  This fixes possible stack overflows
+
+* [#9085] Fix possible deadlock during indexing
+
+* [#9058] Fixes procedural macro expansion on Rust beta
+
+* [#9009] Fix ["Inline function"] refactoring dialog when function has single usage in another module
+
+* [#8951] Do not offer to remove unused variables in places where they are required (by [@Kobzol])
```

Comment:
Actually, it's quite unclear phrase, from my point of view. Let's rephrase it somehow